core/tests/protractor_utils/ExplorationEditorTranslationTab.js

```diff
@@ -216,6 +236,80 @@ var ExplorationEditorTranslationTab = function() {
       'State translation editor takes too long to disappear');
   };
 
+  this.expectSaveUploadedAudioButtonToBeDisabled = function() {
+    expect(saveUploadedAudioButton.getAttribute('disabled')).toBe('true');
+  };
+
+  this.uploadAudio = function(relativePathOfAudioToUpload) {
+    var audioAbsolutePath = path.resolve(
+      __dirname, relativePathOfAudioToUpload);
+    audioUploadInputElement.sendKeys(audioAbsolutePath);
+    waitFor.elementToBeClickable(
+      saveUploadedAudioButton, 'Save button is not clickable');
+    saveUploadedAudioButton.click();
+    waitFor.invisibilityOf(saveUploadedAudioButton,
+      'Upload Audio modal takes too long to disappear');
+  };
+
+  this.expectWrongFileType = function(relativePathOfAudioToUpload) {
+    var audioAbsolutePath = path.resolve(
+      __dirname, relativePathOfAudioToUpload);
+    audioUploadInputElement.sendKeys(audioAbsolutePath);
+    expect(element(by.css('div.error-message')).getText())
+      .toContain('This file is not recognized as an audio file.');
+  };
+
+  this.expectAudioOverFiveMinutes = function(relativePathOfAudioToUpload) {
+    var audioAbsolutePath = path.resolve(
+      __dirname, relativePathOfAudioToUpload);
+    audioUploadInputElement.sendKeys(audioAbsolutePath);
+    waitFor.elementToBeClickable(
+      saveUploadedAudioButton, 'Save button is not clickable');
+    saveUploadedAudioButton.click();
+    waitFor.visibilityOf(audioOverFiveMinutesErrorMessageElement,
+      'Error element is not visible');
+    expect(audioOverFiveMinutesErrorMessageElement.getText()).toContain(
+      'Audio files must be under 300 seconds in length.');
+  };
+
+  this.openUploadAudioModal = function() {
+    waitFor.elementToBeClickable(
+      uploadAudioButton, 'Upload Audio button is not clickable');
+    uploadAudioButton.click();
+  };
+
+  this.closeUploadAudioModal = function() {
+    waitFor.elementToBeClickable(
+      closeAudioUploaderModal, 'Close audio uploader modal is not clickable');
+    closeAudioUploaderModal.click();
+  };
+
+  this.playOrPauseAudioFile = function() {
+    waitFor.visibilityOf(
+      playPauseAudioButton,
+      'Play or pause audio button is taking too long to appear');
+    playPauseAudioButton.click();
+    return this.isAudioPlaying();
+  };
+
+  this.isAudioPlaying = function() {
+    return audioMaterialSliderDiv.getAttribute('aria-valuenow')
+      .then(function(firstValue) {
+        return new Promise(function(resolve, reject) {
+          setTimeout(function() {
+            resolve(firstValue);
+          }, 2000);
+        });
+      }).then(function(firstValue) {
+        return audioMaterialSliderDiv.getAttribute('aria-valuenow')
+          .then(function(secondValue) {
+            return +firstValue !== +secondValue;
```

Comment:
should we handel the null value for the `firstValue` & `secondValue` (considering +null = 0)?

Also, can we add a strict check like `firstValue < secondValue`?

Reply:
Done